pkgs/os-specific/linux/kernel/common-config.nix

```diff
@@ -816,6 +816,12 @@ let
 
       # See comments on https://github.com/NixOS/nixpkgs/commit/9b67ea9106102d882f53d62890468071900b9647
       CRYPTO_AEGIS128_SIMD = whenAtLeast "5.4" no;
+
+      # Distros should configure the default as a kernel option.
+      # We previously defined it on the kernel command line as cma=
+      # The kernel command line will override a platform-specific configuration from its device tree.
+      # https://github.com/torvalds/linux/blob/856deb866d16e29bd65952e0289066f6078af773/kernel/dma/contiguous.c#L35-L44
+      CMA_SIZE_MBYTES = freeform "32";
```

Comment:
Note that the platform specific config is not always good. At least on rpi3, I had to set it to 256M to get plymouth to display properly.

Reply:
Yes, though this becomes the *default*, which was previously set on the command-line. The command-line still has precedence over this value.

It is actually hard to know what value should be used! It seems basically no one is in agreement.

So for the time being I only *moved* the value so the Raspberry Pi 4B can use whatever the device tree defines.

Reply:
Since you can reclaim cma not in use, a higher value may be a good idea. I wish rpi published defaults for it.

Reply:
Yes, multiple angles on that:

 1. Each board vendors should provide a device tree entry with what is needed to make them useful. Probably through an overlay. This is because some boards have *issues* that are specific! E.g. here the Pi4 can't use CMA higher than 1GiB!
 1. We can provide known useful values per-board in some manner. As it seemingly can't be universal (see previous issue)
 1. What even *is* a good universal value to use here?

Right now I really only moved the current value to a location where it has the least precedence. Let's leave *changing* the value to a further PR. I don't want this change set to start breaking things :) (even though I'm pretty sure it would be fine).

Reply:
Adding another note about `cma`. Through the device tree it's entirely possible that *multiple* CMA ranges are allocated, and needed for some devices. I haven't looked at it in details, but I saw that transpire in a Mobile NixOS device.